client/core/wallet.go

```diff
@@ -86,34 +107,37 @@ func (w *xcWallet) refreshUnlock() (unlockAttempted bool, err error) {
 	if !w.Locked() {
 		return false, nil // unlocked
 	}
+
 	// Locked backend requires both encrypted and decrypted passwords.
-	if len(w.encPW()) == 0 {
-		return false, fmt.Errorf("%s wallet reporting as locked but no password has been set", unbip(w.AssetID))
-	}
 	w.mtx.RLock()
-	defer w.mtx.RUnlock()
-	if len(w.encPass) == 0 {
+	pwUnset := len(w.encPass) == 0
+	locked := len(w.pw) == 0
+	w.mtx.RUnlock()
```

Comment:
This is the **un**locking method, but blocking other methods like `(*xcWallet).state` was probably an issue.  With bitcoind and huge numbers of keys, it is reported that Unlock can take several minutes to complete.